src/org/zalando/stups/mint/worker/job/run.clj

```diff
@@ -0,0 +1,63 @@
+(ns org.zalando.stups.mint.worker.job.run
+  (:require [org.zalando.stups.friboo.system.cron :refer [def-cron-component job]]
+            [org.zalando.stups.friboo.log :as log]
+            [org.zalando.stups.friboo.config :as config]
+            [org.zalando.stups.mint.worker.job.sync-app :refer [sync-app]]
+            [org.zalando.stups.mint.worker.external.storage :as storage]
+            [overtone.at-at :refer [every]]
+            [clj-time.core :as time]))
+
+(def default-configuration
+  {:jobs-cpu-count        1
+   :jobs-every-ms         10000
+   :jobs-initial-delay-ms 1000
+   :jobs-max-s3-errors    10})
+
+; used to track rate limiting of service user api
+(def rate-limited-until (atom (time/epoch)))
+
+(defn run-sync
+  "Creates and deletes applications, rotates and distributes their credentials."
+  [configuration tokens]
+  (try
+    (when (time/after? (time/now)
+                       @rate-limited-until)
+      (let [storage-url (config/require-config configuration :mint-storage-url)]
+        (log/debug "Starting new synchronisation run with %s..." configuration)
+
+        (let [apps (storage/list-apps storage-url tokens)]
+          (log/debug "Found apps: %s ..." apps)
+          (doseq [app apps]
+            (try
+              (sync-app configuration
+                        app
+                        tokens)
+              (storage/update-status storage-url (:id app) 
+                                                 {:has_problems false
+                                                  :s3_errors 0
+                                                  :message ""}
+                                                 tokens)
+              (catch Exception e
+                (when (= 429 (:status  (ex-data e)))
+                  ; bubble up if we are rate limited
+                  (throw e))
+                (storage/update-status storage-url (:id app)
+                                                   {:has_problems true
+                                                    :s3_errors (if (= "S3Exception" (:type (ex-data e)))
```

Comment:
But the "body" is only one form, so it doesn’t add anything, really.

    user=> (macroexpand '(when true 0))
    (if true (do 0))

But I’m not religious about it, we can change that.